packages/syft/src/syft/core/smpc/protocol/beaver/beaver.py

```diff
@@ -0,0 +1,289 @@
+"""Beaver Triples Protocol.
+
+D. Beaver. *Efficient multiparty protocols using circuit randomization*.
+In J. Feigenbaum, editor, CRYPTO, volume **576** of Lecture Notes in
+Computer Science, pages 420–432. Springer, 1991.
+"""
+
+
+# stdlib
+import operator
+import secrets
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Tuple
+
+# third party
+import numpy as np
+
+# relative
+from ....tensor.smpc.mpc_tensor import MPCTensor
+from ....tensor.smpc.share_tensor import ShareTensor
+from ...store import register_primitive_generator
+from ...store import register_primitive_store_add
+from ...store import register_primitive_store_get
+from ...store.exceptions import EmptyPrimitiveStore
+
+ttp_generator = np.random.default_rng()
+
+
+def _get_triples(
+    op_str: str,
+    nr_parties: int,
+    a_shape: Tuple[int],
+    b_shape: Tuple[int],
+    ring_size: int = 2 ** 32,
+    **kwargs: Dict[Any, Any],
+) -> Tuple[Tuple[Tuple[ShareTensor, ShareTensor, ShareTensor]]]:
+    """Get triples.
+
+    The Trusted Third Party (TTP) or Crypto Provider should provide this triples Currently,
+    the one that orchestrates the communication provides those triples.".
+
+    Args:
+        op_str (str): Operator string.
+        nr_parties (int): Number of parties
+        a_shape (Tuple[int]): Shape of a from beaver triples protocol.
+        b_shape (Tuple[int]): Shape of b part from beaver triples protocol.
+        ring_size (int) : Ring Size of the triples to generate.
+        kwargs: Arbitrary keyword arguments for commands.
+
+    Returns:
+        List[List[3 x List[ShareTensor, ShareTensor, ShareTensor]]]:
+        The generated triples a,b,c for each party.
+
+    Raises:
+        ValueError: If the triples are not consistent.
+        ValueError: If the share class is invalid.
+    """
+    # relative
+    from ..... import Tensor
+
+    cmd = getattr(operator, op_str)
+    min_value, max_value = ShareTensor.compute_min_max_from_ring(ring_size)
+    seed_shares = secrets.randbits(32)
+
+    a_rand = Tensor(
```

Comment:
I'm pleasantly surprised to see Tensor working so well like this. I hadn't put as much thought into internal uses of Tensor objects (as opposed to using numpy arrays or something). i'm glad you did this and i'm glad it's working.